Scoutsuite docs (#1013)
guardicore/monkey: docs/content/reference/zero-trust/scoutsuite.md

---
title: "Scoutsuite"
date: 2021-03-02T16:23:06+02:00
draft: false
---

### What is ScoutSuite?

<a href="https://github.com/nccgroup/ScoutSuite" target="_blank" >Scout Suite</a> is an open-source cloud security-auditing tool.

> It queries the cloud API to gather configuration data of the cloud infrastructure. Based on configuration

Reviewer: ```suggestion
It queries the cloud API to gather configuration data from the cloud infrastructure. Based on configuration
```

Author: This sentence tries to say that the data gathered is about cloud infrastructure, not that it's gathered from cloud infrastructure. Fixed to `It queries the cloud API to gather configuration data about the cloud infrastructure.`

> data gathered ScoutSuite shows security issues and risks present in your infrastructure.

Reviewer: ```suggestion
data gathered, ScoutSuite shows security issues and risks present in your infrastructure.
```

### Which cloud providers are supported?

> So far the Infection Monkey only supports AWS.

Reviewer: ```suggestion
Today, the Infection Monkey only supports AWS.
```

Author: Fixed to `Currently`, as we refer to this point in time, not a particular time unit. Could be removed altogether though

> ### How to enable ScoutSuite?

Reviewer: ```suggestion
### How to enable ScoutSuite
```

Author: Reworded all headers

> First of all, Infection Monkey needs access to your cloud API. You can provide access

Reviewer: ```suggestion
First, Infection Monkey needs access to your cloud API. You can provide access
```

in the following ways:

 - Provide access keys:
    - Create a new user with ReadOnlyAccess and SecurityAudit policies and generate keys
    - Generate keys for your current user (faster but less secure)
 - Configure AWS CLI:
    - If the command-line interface is available on the Island, it will be used to access 
    the cloud API
    

> More details about configuring ScoutSuite can be found in the tool itself, by choosing 

Reviewer: Maybe add a screenshot that shows the Run Monkey page as it appears today.

"Cloud Security Scan" in the "Run Monkey" options. 

> After you're done with the setup, make sure that a checkmark appears next to the AWS option to 
> verify that ScoutSuite can access the API.

Reviewer: ```suggestion
After you're done with the setup, make sure that a checkmark appears next to the AWS option. This 
verifies that ScoutSuite can access the API.
```

![Successfull setup indicator](/images/reference/zero-trust/scoutsuite_aws_configured.png 
"Successful setup indicator")

> ### How to run cloud scan?

Reviewer: ```suggestion
### How to run cloud security scan
```

> If you have successfully configured cloud scan, once the Monkey Agent is run **on the Island**, 
> the cloud infrastructure will get scanned. To make this happen, you can simply click on "From Island" 
> in the run options. The scope of network scan and other activities you configured the Agent to 
> do are irrelevant for cloud security scan, except 
> **Monkey Configuration -> System info collectors -> AWS collector** which needs to remain **enabled**.

Reviewer: ```suggestion
If you have successfully configured the cloud scan, Infection Monkey will scan
your cloud infrastructure when the Monkey Agent is run **on the Island**. You
can simply click on "From Island" in the run options to start the scan. The
scope of the network scan and other activities you may have configured the Agent
to perform are ignored by the cloud security scan, except **Monkey
Configuration -> System info collectors -> AWS collector**, which needs to
remain **enabled**.
```

> ### How to assess cloud scan results?

Reviewer: ```suggestion
### How to assess cloud scan results
```
